### ISSUE TITLE
⭐ [TESTERS NEEDED] v1.14.0 Beta

### ISSUE THREAD

**Issue by xanderfrangos** [xanderfrangos/twinkle-tray#296]
Hi all,

The next version of Twinkle Tray (v1.14.0) has had significant changes since v1.13.11. I'm looking for people to help test out this update and find any major issues before release.

Here's the current changelog for v1.14.0:

> **New features & improvements:**
> - Added a Windows 11 UI, available as an option for both Windows 10 & 11 users. Additionally, there have been several small tweaks to the UI to better reflect the Windows 10/11 UI.
> - Added settings to show contrast, volume, and power options for supported displays. You can find these features in the new "DDC/CI Features" settings menu.
> - Added an option for idle detection. Your displays can automatically dim after a set amount of time.
> - Added an option to hide your laptop/tablet's display from the brightness flyout when closed or disabled.
> - Updated process for reading monitor info to improve accuracy and responsiveness.
> - Restored WMIC code from v1.13.10 and improved handling of WMIC/WMI failures. (Special thanks to @DustRaider8 and @rp1231 for helping debug the issue.)
> 
> **Other information:**
> - By default, Twinkle Tray no longer gets the latest brightness for DDC/CI displays. This is to prevent issues with some displays and reduce visual glitches. If you would like to restore the old behavior, it's available in the "DDC/CI Features" menu in Settings.
> - I have chosen to go with an emulated "Mica" material over "Acrylic" for Windows 11, as implementing Acrylic with rounded corners is currently impossible in Electron.
> - The "Mica" effect in the Windows 11 UI currently only works on the main display. It also does not correctly detect slideshow wallpapers. If you use slideshow wallpapers, I recommend turning off the Mica effect.

If you'd like to help out, you can download the v1.14.0 beta here:
[Download Twinkle Tray v1.14.0-beta7](https://1drv.ms/u/s!AiWKFpjcSKRQhtdyWUdtZuawLLzNxw?e=XM6Ft5)

Please let me know if you run into any issues or weirdness with it. If you do try out the beta, please subscribe to this issue so I can notify you of updates. Thanks!

**Comment by xanderfrangos** [comment]
I've noticed an issue on my laptop where the "Disable WMIC/WMI" toggles kicked in when they shouldn't, making my laptop display not list in Twinkle Tray. If you encounter this as well, you can disable these in the **Settings -> Debug** menu.

<img width="768" alt="Screenshot 2022-01-02 200059" src="https://user-images.githubusercontent.com/33106561/147893858-aa186d4a-9b81-439e-b448-a5c5f720e942.png">

**Comment by DustRaider8** [comment]
Regarding the issues from the previous version, all seems good on my end. 
However, I just stumbled upon an old issue that I've encountered on previous versions and I totally forgot about it. I will open a separate issue for it because it's not related to this and it goes way back.

**Comment by rp1231** [comment]
Everything seems to be working fine for me as well.
And the new windows 11 theme looks great!
Will let you know if I encounter any issues after extended usage.

**Comment by rp1231** [comment]
Also one suggestion would be that the windows 11 style should be enabled by default on the windows 11 OS.
I knew it existed because I pay attention to the changelogs.
But I think most users would not even know it exists, which would be a shame.
And I think that most users would prefer consistency in their OS  anyways.

**Comment by DustRaider8** [comment]
@rp1231 For me the Windows 11 style is the default on the beta version posted above. Can you make sure whether you have this "v1.14.0-beta1" and not the "v1.14.0-preview8"?

**Comment by rp1231** [comment]
@DustRaider8 
Yes I have the V1.14.0-beta1 installed...

I had to change the theme manually.

**Comment by DustRaider8** [comment]
@rp1231 Ok, so I tried updating from v1.14.0-preview8 to "v1.14.0-beta1 without uninstalling v1.14.0-preview8 first and yes, Windows 10 style is still selected in the settings menu. This isn't really a bug, it just sticks to the settings you had before and that's what should happen. If you clean install the newest version by uninstalling the previous, you will get the Windows 11 style as default for sure.

**Comment by rp1231** [comment]
I would still consider this a semi bug as many people will be upgrading from the previous stable version
(which did not have the option of the windows 11 style)
In which case the settings from the previous version will be carried on.
Which is what happened to me.

Not really a major issue.
It's just that I would have missed this option if I hadn't gone through the change logs.

**Comment by DustRaider8** [comment]
@rp1231 Going from the latest v1.13.11 to v1.14.0-beta1 also sets the style to default which enables W11 style if you're on W11 OS. And that's what matters at the end.

**Comment by xanderfrangos** [comment]
@rp1231 There was an override in the test version you were using before that forced the Windows 10 style on because the Windows 11 style wasn't fully ready yet. So it was likely still set to the Windows 10 style for you. Users upgrading from 1.13.X to 1.14.0 won't have that issue.

**Comment by LupusHacker** [comment]
The brightness slider from 1.13.0 seems to be missing, other than that It's actually working compared to 1.13.11.

Recommendations/Notes:

- I know you're trying to stick with the Windows theme, but I'd rather type the scheduled times then scroll through them, also making it clear when that time resets (is it at midnight?, does it only change if there is another time scheduled to "interrupt it"?, etc.)
- The brightness flickers when I left-click the tray icon. (Isn't this supposed to bring up a tray showing the displays current brightness?)

**Comment by LupusHacker** [comment]
Wait, are we supposed to do a clean install?

**Comment by xanderfrangos** [comment]
@LupusHacker Thanks for the list. I'll look into these when I can. 👍

**Comment by Ivan-Malinovski** [comment]
Just installed from the link in the initial post; it seems to work fine, and it fixed the issue I had with my displays not being recognized in 1.13.11. I had to restart to make it recognize my MSI monitor, but after that, it was all smooth sailing!

**Comment by rp1231** [comment]
I don't know if anyone else has been having this problem,
but the brightness panel takes a few clicks to pop up , especially the first time I click on the tray icon.
Either that or there is a huge lag before it shows up.
After the first delayed pop up it seems to work fine.

Also wanted to ask if you could potentially change the behavior of the pop up so that
when you click on the tray icon, it closes if it's already open.
Other wise it kind of flickers awkwardly and that feels a little off.
I think that the pop up closing when clicked on a second time would also be consistent with
how pop up panels work in windows in general.

Thanks

**Comment by TheShark27** [comment]
I think as a beta this looks extremely good, but I had two ideas of possible improvements upon using it.
1. Windows 10/11 allow the user to choose whether or not the OS' accent color applies to Start and Taskbar (and also other surfaces like the Action Center, notification, calendar). Your design already includes it, and it looks very good, but I think it would be great customization-wise to be able to disable it and let it stick to a white/black theme (with the same transparency effect as Mica though).
2. Since I started using your software I've always noticed the absence of closing animations for both the menu and the keyboard shortcut brightness overlay. Would it be possible to add them?

I also have an extra suggestion.
Microsoft released a new build a few days ago with a new volume slider menu. It's the one that appears when using keys to adjust the volume level. Having a similar design for the brightness slider would be very fitting.
However, that Windows version is expected to release later this year, so that design might change drastically and it might not be worth the effort right now.

Thanks for your work on this awesome utility, and for letting us test this beta!

**Comment by ArthurWolfhound** [comment]
Hi, thank you for that new beta. I am on Windows 11, and glad to say that everything looks (and works) great right now.

## Monitor Settings & DDC/CI Features
How do "Brightness update rate" and "Get Current Brightness" correlate with each other? I mean, I can understand the former - explanation is good, but what is the deal with latter? It says that it will always get the brightness level from the monitor when opening a `brightness page`... what page?

In the same basket - can we have current brightness sliders within the application? It would help when testing (reasons in the Turn off feature problems below)

## DDC/CI Features
### Contrast controls
Is it possible to detect that monitor does not respond, or specifically rejects "contrast change" to show notification to the user? Not really important, just for the sake of UI/UX - had to unlock "Manual" mode in my monitor menu for it to start reacting to the contrast slider. 

### Volume controls
If you change to the "linked brightness control" - volume control for my external display vanishes, which makes the feature useless.

### Turn Off Display
If I use "turn off" display with my LG 27UP850-W (LG HDR 4K) it seems to turn off, because if I use power button on it - it shows OSD which only shows on turn on sequence. But Windows does not register display disconnection - which is kinda weird.

**_Reasons for the current brightness sliders within settings_**: When turned off - my `main, external` monitor remains soft-locked connected, and I can't access the pop-up to check with the brightness sliders.

## Hotkeys & Shortcuts
Are you sure that "Turn off display action" drop-down menu should be under "Hotkeys & Shortcuts"? I feel it would be logical to move it under "DDC/CI Features" where the "Turn off display action" is actually being turned on?

**Comment by DustRaider8** [comment]
I decided to play PUBG and it's consistently crashing a few minutes into the game. If I exit out of Twinkle Tray, it stops crashing completely. It's also crashing on v1.13.10. I've never played PUBG before so I can't know if the problem with the game is recent or not. Also, I'm not sure if it's anti-cheat related or something else because the game's crash reporter just says "Uknown crash".
This is the only game I've encountered this problem with and I'm not expecting a fix, just posting this to keep it in mind.

**Comment by TheShark27** [comment]
> How do "Brightness update rate" and "Get Current Brightness" correlate with each other? I mean, I can understand the former - explanation is good, but what is the deal with latter? It says that it will always get the brightness level from the monitor when opening a `brightness page`... what page?

I'm not the developer, but I think I understand the difference between the two settings. The first basically controls how big the "steps" between each brightness level are. That only occurs when you manually update the brightness though.

The other setting is also pretty well explained in the settings page. If enabled, whenever you open the brightness page (which, to my understanding, is the normal menu on the bottom right), the app will "ask" the monitor what the current brightness is, and adjust its slider accordingly. It's pretty much useless if you always use Twinkle Tray to change your brightness, but it can be useful if you also use another app to change the monitor's brightness, the default Windows slider you find in laptops, or the monitor settings themselves. 

If you leave it disabled and change the brightness without using Twinkle Tray, the app would have no way of knowing the new brightness and would report the last one you set it to, which would be inaccurate.

**Comment by ArthurWolfhound** [comment]
@TheShark27 thanks!

**Comment by rp1231** [comment]
@xanderfrangos 
I've been getting an issue detecting my laptop screen after waking it up from sleep.
My external monitor gets detected, but the slider for the laptop screen is missing after waking it up.

Restarting the app gets the app to detect the laptop screen again.

**Comment by Linx-ESP** [comment]
> I decided to play PUBG and it's consistently crashing a few minutes into the game. If I exit out of Twinkle Tray, it stops crashing completely. It's also crashing on v1.13.10. I've never played PUBG before so I can't know if the problem with the game is recent or not. Also, I'm not sure if it's anti-cheat related or something else because the game's crash reporter just says "Uknown crash".
> 
> This is the only game I've encountered this problem with and I'm not expecting a fix, just posting this to keep it in mind.

Same happened to me in 1.13, with 1.14 a message appears after crashing
![Captura de pantalla (181)](https://user-images.githubusercontent.com/51830853/150441198-88f11da2-0906-48c6-8e4d-bbf7e28f010b.png)

**Comment by DustRaider8** [comment]
@Linx-ESP Damn. For you, the anti-cheat actually tells that it's blocking Twinkle Tray. For me, the game is just crashing and crash reporter pops up instead of shield reporter and it just says "Unknown crash". I had to check everything before realizing Twinkle Tray is the problem.
![Screenshot 2022-01-14 010941](https://user-images.githubusercontent.com/87049956/150444535-4c772905-9f5b-42f9-b252-d3f1ff905d20.jpg)

**Comment by pc-v2** [comment]
![image](https://user-images.githubusercontent.com/58063311/150638893-a4bb29bd-912a-450d-9a40-27fc8974843a.png)
my pc can't sleep with 1.14

**Comment by yume-chan** [comment]
I'm using a LG 27GL850 monitor and I found two issues:

1. Mouse wheel scrolling doesn't work when hovering on the volume slider. It works when hovering on brightness slider.
2. If I change the volume, then close and re-open the popup window, it shows the last volume value, instead of current one. (It correctly set volume on my monitor, as the monitor shows its OSD when volume was changed via DDC/CI). After multiple close and re-open, it will finally show the correct volume value.

Another small problem:

Clicking the tray icon twice should hide it (like all native Windows popups), instead of playing the slide in animation again.

**Comment by J0n4t4n** [comment]
Just installed, because I was looking for the Volume control feature.
It works very well and is super useful.

The one thing I am missing is Hotkeys for the volume.
I would love if it could override the default Windows volume buttons, so I could control the monitor volume with my keyboard buttons, like I can with MonitorControl on my Mac.

**Comment by SipanDeerns** [comment]
Works also very well for me. Cool design. The only thing is if my monitor is in sRGB preset the monitor doesnt allow me to change brightness (feature of the monitor). It would be nice to change the presets of the monitor so I can adjust the brightness.

**Comment by sipaan** [comment]
Works also very well for me. Cool design. The only thing is if my monitor is in sRGB preset the monitor doesn't allow me to change brightness (feature of the monitor). It would be nice to change the presets of the monitor so I can adjust the brightness.

**Comment by veehexx** [comment]
been using this for a few days and so far it's been reliable on Win11.

what i have found is when unpluging or changing displays that Twinkle Tray will not detect the changes in the taskbar icon, until the main settings window has been opened.

so for laptop use: i connect a screen. The tray icon shows pre-connected displays (so internal laptop only). i have to go into the settings for twinkle tray to redetect the displays which the try icon then shows the internal & external display.
same for disconnecting; the tray icon shows the previously-connected displays. i have to go into settings to redetect which the tray icon will then remove the external screens.

i use both a usb-C portable adaptor, and a Thunderbolt docking station depending on my location; both with external screens connected rather than direct HDMI/DP connection.

There isnt any reasonable timeperiod when this seems to automatically occur either so either thats completely broken, or we need a new feature for periodic redetecting.

**Comment by mipak** [comment]
first thanks for the great software, it's really a life saver :-)

I'm using a Huawei MateView 28" 4K screen and currently testing the beta, especial for the contrast feature.

* I adjust the contrast from 50 (default) to 40 and close the slider (works).
* The real contrast of the monitor is now at 40 (works). 
* I open the slider again, contrast slider is back at 50 (real contrast is still 40). 
* If I click on the slider to adjust it again the real contrast of the monitor jumps to 50, the moment I click on the slider at 50.

So it seems when the contrast slider is closed it does not remember the setting it was left at, but if..

* I adjust the contrast from 50 (default) to 40 and close the slider (works).
* The real contrast of the monitor is now at 40 (works). 
* I open the slider again, contrast slider is at 50 again, but I do not touch it and close the slider again.
* I open the slider again, contrast slider is at 40 (correct).

So it seems at some point it sometimes forgets to remember the set value, when it's set. I enabled/disabled the "always get current brightness", but it did not change anything.

feature request on the side:
please make contrast also available at time based settings, once its enabled at DDC.

thanks so much!

**Comment by xanderfrangos** [comment]
Hi all,

Thank you for the feedback so far! I haven't had much time to work on Twinkle Tray lately, but I'm still monitoring this thread and tracking bugs as you report them.

**Comment by itsforrest** [comment]
Just found out that you started implementing a volume feature! Love it!

Will you be **enable hotkeys for volume?** And would it be possible to enable the user(me 😅) to **disable the brightness slider** and only show the volume slider in the tray menu?

Anyway, thank you for creating this amazing piece of software!

**Comment by pc-v2** [comment]
got a bug for one of my display,
my display:
LG 27GP850 (Main Display) <--- the bugged one 
LG 22MP68V-Q (Secondary Display)
HUION KAMVAS 16 Pro (For Drawing)
I need to retoggling Get Current Brightness to fix it

[settings.txt](https://github.com/xanderfrangos/twinkle-tray/files/8368224/settings.txt)
[raw monitor data.txt](https://github.com/xanderfrangos/twinkle-tray/files/8368232/raw.monitor.data.txt)

**Comment by pc-v2** [comment]
disabling win32 seems fixed it

**Comment by fhoshino** [comment]
Is there a command line flag for contrast control?

**Comment by GieltjE** [comment]
Been testing this release for more than 3 weeks now on the latest windows 11 insider build(s).
So far everything has been problem free and function as good as we can wish.

**Comment by xanderfrangos** [comment]
Hi all,

Thank you for your patience. I finally have a new beta version prepared (**v1.14.0-beta2**). See the [initial post](https://github.com/xanderfrangos/twinkle-tray/issues/296#issue-1092070568) for the download link.

**The changes since beta-1 include:**
- Fixed Twinkle Tray preventing PCs from sleeping.
- Updated dependencies (potentially fixing various issues).
- Imported latest localization files.
- Adjusted timings for reading current monitors after sleep and hardware changes. It may take up to 10 seconds for Twinkle Tray to find your displays after one of these events.
- Added "Refresh monitors" to the tray icon context menu, for in case your displays are intermittently not showing up. If you're seeing a pattern in your display(s) not showing up, please still let me know so I can look into it. 
- Adjusted how the Twinkle Tray UI displays over the current window to (hopefully) fix the many reported issues.
- Adjusted "Idle Detection" behavior to (hopefully) work more reliably when waking up displays.
- Added the `--List` flag to the command line to list current displays. 
- Made Time of Day adjustments slower/smoother. 
- Adjusted automatic disabling of WMI reading to avoid false positives (ex. waking up from sleep).

At this point I'm just hoping to get confirmation from you guys if everything is good with this version so I can publish it. I've seen a bunch of requests about features such as:
- Contrast/Volume control via hotkeys/time-of-day adjustments/command line.
- Windows 11-style hotkey overlay.
- Replacing brightness slider with other DDC/CI features.

However, those won't be making it into 1.14.0. I've been sitting on this long enough. There are a lot of important improvements/changes for Windows 11 users that I can't put off any longer. You all have provided a ton of valuable feature suggestions that I plan to address in 1.15.0+.

Again, let me know how this new beta is working and hopefully I can get this out (finally 😅).

Thanks!

**Comment by rp1231** [comment]
Working fine for me, but one minor bug is that scrolling over the contrast slider doesn't change the value.
Where as it scrolling to change the value works for the brightness sliders.

**Comment by rp1231** [comment]
Also the popup seems to be very buggy for me.
It works sometimes and mostly doesn't.
It works reliably only when I disable both wmi and wmic.
But if I disable those two options, the internal display isn't detected(Even if the Win32 option is active)

This was happening in the previous beta as well.

**Comment by mipak** [comment]
unfortunately my contrast slider update issue still exists, as described [above #issuecomment-1035894696](https://github.com/xanderfrangos/twinkle-tray/issues/296#issuecomment-1035894696)

**Comment by rp1231** [comment]
@xanderfrangos 

> When you say "Also the popup seems to be very buggy for me", do you mean it doesn't list any displays or it doesn't display anything at all? Also, FYI you need WMI or WMIC to show your internal display. If you're on Windows 11 Build 22621+, [try re-enabling WMIC as an optional feature](https://www.elevenforum.com/t/add-or-remove-wmic-command-feature-in-windows-11.5119/) and make sure you haven't disabled WMIC in Twinkle Tray. Let me know if that helps.

It doesn't display anything at all.....
I tried the guide that you posted and I don't seem to have the option to install wmic, so unfortunately that didn't work for me.....
The best way that it works is if I disable all of the three options(WMIC,WMI,Win32).

Also the older version of twinkle tray is working fine for both the displays, so maybe the issue isn't with wmi not existing on my pc.

The windows 11 build I'm using is 220000.778

**Comment by rp1231** [comment]
> for some reason the current beta doesn't always pop up on the first tray icon click.. sometimes I need to click 3 times until the sliders show... but only sometimes. most of the time it works at first click. wasn't the case in the previous beta.. I think.. at least I never noticed that behavior before. using windows 10 pro.

I've been experiencing this as well.

**Comment by xanderfrangos** [comment]
@mipak @rp1231 I've uploaded a new beta that should address the flyout not appearing when clicking. Let me know if it helps. I've also fixed the bug where volume/contrast doesn't stick after clicking away, as well as the scroll wheel not applying to those sliders.

**Comment by mipak** [comment]
@xanderfrangos fix confirmed. works for me now. thank you :-)

**Comment by rp1231** [comment]
@xanderfrangos the fixes are working fine for me as well.
But the wmi issue remains.
If I have any of the wmi options enabled, the pop up doesn't show up at all or doesn't detect any display(depending on the combination of the disabled features)

Thanks for the fixes so far!

**Comment by rp1231** [comment]
@xanderfrangos the fixes are working fine for me as well.
But the wmi issue remains.
If I have any of the wmi options enabled, the pop up doesn't show up at all or doesn't detect any display(depending on the combination of the disabled features)

Thanks for the fixes so far!

**Comment by rp1231** [comment]
@xanderfrangos the fixes are working fine for me as well.
But the wmi issue remains.
If I have any of the wmi options enabled, the pop up doesn't show up at all or doesn't detect any display(depending on the combination of the disabled features)

Thanks for the fixes so far!

**Comment by rp1231** [comment]
@xanderfrangos the fixes are working fine for me as well.
But the wmi issue remains.
If I have any of the wmi options enabled, the pop up doesn't show up at all or doesn't detect any display(depending on the combination of the disabled features)

Thanks for the fixes so far!

**Comment by xanderfrangos** [comment]
Hi all,
The first post has been updated with *beta4*, which is hopefully the final one before launch. 🤞

I've added Troubleshooting section to the Settings window, adjusted how the command line feature works internally, and added a fallback for setting the brightness when WMI-Bridge (previous referred to as just *WMI*) is disabled. I'm also working on updating the readme and wiki for v1.14.0.

@rp1231 The new troubleshooting section is the best I can do for other users that are encountering the same issues as you. I'm out of ideas, unfortunately. Once Windows 11 22H2 officially releases, I would recommend re-enabling WMIC per my previous instructions and see if that works at all. But if you're not using a laptop, disabling all features should be OK in theory. Some stuff won't work like that (like name detection), but the barebones features should. Older versions of Twinkle Tray only used WMIC.

**Comment by rp1231** [comment]
Will do.

**Comment by rp1231** [comment]
@xanderfrangos 

I don't think that the problem isn't wmi as:
1. As per this article https://winaero.com/microsoft-removed-the-wmic-tool-from-windows-11-dev-builds/amp/
wmic.exe and wmic.exe.mui  should be present in their respective folders for wmi to work on your system. 
Both of these files are present in my system.


2. Wmi has only been removed from the dev builds and I'm on the normal build.

3. Other apps that use wmi work perfectly fine( like monitorian and midi-mixer with the brightness plugin and the older version of twinkle tray itself)

Therefore the re-enabling wmi solution is not relevant for the above stated reasons(as it's already enabled).....

Furthermore I came across an interesting observation during the installation of the latest version of the program (beta4).
I hit the reset button and both the monitors showed up.
But the problem started again when I enabled the contrast slider in the "DDC/CI features".
I tried this multiple times and the same thing happened (resetting the settings and then enabling the contrast slider).

This leads me to believe that there is something in the implementation of the contrast slider that is causing the problems....
Another reason why I think this to be true is that these problems started popping up only after the implementation of the contrast slider........
Before this there were no issues with twinkle tray.....

These are just guesses but I'm thinking that this may be the root cause of the problem.....

**Comment by rp1231** [comment]
@xanderfrangos 

The problem doesn't resolve itself by turning of the contrast slider, but it does resolve itself after I turn it off and restart the app.

Here's the raw monitor data when it's not working:
`
{
  "4&160c887b&2&UID265988": {
    "id": "\\\\?\\DISPLAY#LGD046F#4&160c887b&2&UID265988",
    "key": "4&160c887b&2&UID265988",
    "num": 0,
    "brightness": 0,
    "brightnessMax": 100,
    "brightnessRaw": 0,
    "type": "wmi",
    "connector": "unknown",
    "min": 0,
    "max": 100,
    "hwid": [
      "DISPLAY",
      "LGD046F",
      "4&160c887b&2&UID265988"
    ],
    "name": "Display 1",
    "serial": "0"
  },
  "4&160c887b&2&UID206395": {
    "id": "\\\\?\\DISPLAY#DELA0B5#4&160c887b&2&UID206395",
    "key": "4&160c887b&2&UID206395",
    "num": 1,
    "brightness": 100,
    "brightnessMax": 100,
    "brightnessRaw": 100,
    "type": "ddcci",
    "connector": "hdmi",
    "min": 0,
    "max": 100,
    "hwid": [
      "\\\\?\\DISPLAY",
      "DELA0B5",
      "4&160c887b&2&UID206395",
      "{e6f07b5f-ee97-4a90-b076-33f57bf4eaa7}"
    ],
    "name": "DELL S2415H",
    "serial": "8CN3Y4BI0M8L",
    "features": {
      "luminance": [
        100,
        100
      ],
      "brightness": false,
      "contrast": [
        100,
        100
      ],
      "powerState": [
        1,
        255
      ],
      "volume": false
    },
    "brightnessType": 16,
    "brightnessValues": [
      100,
      100
    ]
  }
}
`
Note: I compared the raw data when from when it is working and not working, and the raw data during both times is identical.

**Comment by rp1231** [comment]
I have my laptop screen off at the moment, which is why it might show "0" as a value.

**Comment by xanderfrangos** [comment]
@rp1231 Huh, sounds like something is breaking pretty badly... I've built a version of Twinkle Tray with extra logging enabled here: https://1drv.ms/u/s!AiWKFpjcSKRQhtdtBQvOsTSv2G1ctg?e=OHvEPZ

Once you've installed that version, can you start Twinkle Tray from the command prompt (`"%LocalAppData%\Programs\twinkle-tray\Twinkle Tray.exe"` should start it) and paste the log here? Have the contrast slider enabled and try clicking the tray icon before grabbing the log. Hopefully that'll be enough to figure out where it's busted.

**Comment by rp1231** [comment]
I did that and just copied the text from cmd.
Is there a more sophisticated way to get the log?

Anyways, I got a few errors when I clicked on the tray icon with the contrast slider enabled.
Have attached the log file.
The most relevant errors are at line 22, 75 and 97
[log.txt](https://github.com/xanderfrangos/twinkle-tray/files/9048082/log.txt)
.

**Comment by xanderfrangos** [comment]
@rp1231 Give this version a try and let me know if it resolves the contrast slider issue: https://1drv.ms/u/s!AiWKFpjcSKRQhtducj0NqeuFJljBlA?e=syFrdv

**Comment by rp1231** [comment]
Yep, this one's working great!

**Comment by GieltjE** [comment]
When the pc resumes after sleep or when the monitor returns from sleep (power saver mode), the monitor brightness setting is applied again (e.g. monitor is set 33% but twinkletray to 100%, after resume the monitor is on 33% again).
Is this a new bug or is this intended behaviour?

**Comment by pc-v2** [comment]
Even though I have disabled idle timeout, it still dimming after 60sec 
![image](https://user-images.githubusercontent.com/58063311/177968894-9d9f7694-e745-4514-ab9f-5c9790d26f83.png)
![image](https://user-images.githubusercontent.com/58063311/177968920-badf237f-39fa-4573-939d-522796dd082c.png)
![image](https://user-images.githubusercontent.com/58063311/177968906-bc9f66a6-f7d2-4f9a-9e8d-d53386b7a0ff.png)

Is this a new method for idle detection? I've got a bit confusion here since updating to beta 5

```
{
  "isDev": false,
  "userClosedIntro": true,
  "theme": "default",
  "icon": "fluent",
  "updateInterval": 500,
  "openAtLogin": true,
  "brightnessAtStartup": true,
  "killWhenIdle": false,
  "remaps": {},
  "hotkeys": {
    "all__dir1": {
      "monitor": "all",
      "accelerator": "Control+Up",
      "direction": 1,
      "active": true
    },
    "all__dir-1": {
      "monitor": "all",
      "accelerator": "Control+Down",
      "direction": -1,
      "active": true
    }
  },
  "hotkeyPercent": 5,
  "adjustmentTimes": [
    {
      "brightness": 30,
      "hour": 7,
      "minute": 30,
      "am": "AM",
      "monitors": {}
    },
    {
      "brightness": 40,
      "hour": 12,
      "minute": 30,
      "am": "PM",
      "monitors": {}
    },
    {
      "brightness": 20,
      "hour": 6,
      "minute": 30,
      "am": "PM",
      "monitors": {}
    }
  ],
  "adjustmentTimeIndividualDisplays": false,
  "checkTimeAtStartup": true,
  "order": [],
  "monitorFeatures": {
    "GSM5BD3": {
      "powerState": false,
      "contrast": false
    }
  },
  "checkForUpdates": true,
  "dismissedUpdate": "v1.13.11",
  "language": "system",
  "settingsVer": "v1.14.0-beta5",
  "names": {},
  "analytics": true,
  "scrollShortcut": true,
  "useAcrylic": true,
  "useNativeAnimation": false,
  "sleepAction": "ps_ddcci",
  "hotkeysBreakLinkedLevels": true,
  "enableSunValley": true,
  "isWin11": true,
  "windowsStyle": "win11",
  "hideClosedLid": false,
  "getDDCBrightnessUpdates": true,
  "detectIdleTime": "0",
  "disableWMIC": false,
  "disableWMI": true,
  "disableWin32": false,
  "autoDisabledWMI": true,
  "disableOverlay": false,
  "uuid": "45e9e737-3aac-4122-827b-9093909a3e44",
  "branch": "master"
}
```

**Comment by pc-v2** [comment]
it's still dimming when watching videos, even with disabling idle detection

**Comment by pc-v2** [comment]
solved: i need to use wmic

**Comment by fhoshino** [comment]
beta5 gives me CPU usage spikes especially when using chrome, have no idea why, going back to 1.13 resloves the problem.

**Comment by fhoshino** [comment]
`{
  "isDev": false,
  "userClosedIntro": true,
  "theme": "dark",
  "icon": "icon",
  "updateInterval": 500,
  "openAtLogin": true,
  "brightnessAtStartup": true,
  "killWhenIdle": false,
  "remaps": {
    "\\\\?\\DISPLAY#BNQ78E1#4&38b8daf7&0&UID50727680#{e6f07b5f-ee97-4a90-b076-33f57bf4eaa7}": {
      "min": 0,
      "max": 100
    }
  },
  "hotkeys": {},
  "hotkeyPercent": 10,
  "adjustmentTimes": [],
  "adjustmentTimeIndividualDisplays": false,
  "checkTimeAtStartup": true,
  "order": [
    {
      "id": "\\\\?\\DISPLAY#BNQ78E1#4&38b8daf7&0&UID50727680#{e6f07b5f-ee97-4a90-b076-33f57bf4eaa7}",
      "order": 0
    },
    {
      "id": "DISPLAY\\LEN40A2\\4&amp;38b8daf7&amp;0&amp;UID67568640_0",
      "order": 1
    }
  ],
  "monitorFeatures": {
    "BNQ78E1": {
      "contrast": true
    }
  },
  "checkForUpdates": false,
  "dismissedUpdate": "",
  "language": "system",
  "settingsVer": "v1.14.0-beta6",
  "names": {},
  "analytics": false,
  "scrollShortcut": true,
  "useAcrylic": false,
  "useNativeAnimation": false,
  "sleepAction": "ps",
  "hotkeysBreakLinkedLevels": true,
  "enableSunValley": true,
  "isWin11": false,
  "windowsStyle": "win10",
  "hideClosedLid": false,
  "getDDCBrightnessUpdates": false,
  "detectIdleTime": 0,
  "disableWMIC": false,
  "disableWMI": false,
  "disableWin32": false,
  "autoDisabledWMI": false,
  "disableOverlay": false,
  "uuid": "99d0e2b8-b879-4b3a-8ea3-97baf43c8b27",
  "branch": "master",
  "features": {
    "luminance": true,
    "brightness": true,
    "gain": false,
    "contrast": false,
    "powerState": false,
    "volume": false
  },
  "linkedLevelsActive": false
}`

**Comment by xanderfrangos** [comment]
@pc-v2 I actually did not fix the idle issue, so now there's a beta7. I guess this is what happens when I try to program before coffee. 😅

**Comment by xanderfrangos** [comment]
Hi all,
I've released v1.14.0 to GitHub for non-Windows Store users. I'm sure there are still a few bugs to work out, but I'm hoping having a larger pool of users will help get them identified sooner. Feel free to use this issue to post any bugs you encounter with the release version of v1.14.0. I'm going to hold off on uploading to the Windows Store for at least a few days in case there's something serious that slipped through.

@fhoshino I haven't been able to reproduce the issue, but hopefully if it's a common enough issue I'll get feedback from more people to figure out why you're experiencing the CPU spike.

**Comment by xanderfrangos** [comment]
v1.14.3 is now available for all Windows Store users! I think it's finally time to close this issue. Thanks again for your help and patience, everyone.
